### ISSUE TITLE
CI fails Python 2.7 test_pdb.py::test_conect2ter

### ISSUE THREAD

**Issue by orbeckst** [MDAnalysis/mdanalysis#1915]
After PR #1912 was merged we suddenly got the following failure in the Python 2.7 tests only (see 
example run Travis-ci.com [81.3](https://travis-ci.com/MDAnalysis/mdanalysis/jobs/125762063#L1997) from PR #1815) and https://github.com/MDAnalysis/mdanalysis/pull/1912#issuecomment-391654970 :
```
=================================== FAILURES ===================================
_______________________________ test_conect2ter ________________________________
[gw0] linux2 -- Python 2.7.15 /home/travis/miniconda/envs/test/bin/python
    def test_conect2ter():
        def parse():
            with PDBParser(PDB_conect2TER) as p:
                struc = p.parse()
            return struc
    
        with pytest.warns(UserWarning):
>           struc = parse()
E           Failed: DID NOT WARN. No warnings of type (<type 'exceptions.UserWarning'>,) was emitted. The list of emitted warnings is: [].
testsuite/MDAnalysisTests/topology/test_pdb.py:114: Failed
```

Suddenly a warning is *not* emitted anymore.

This failure holds up merging of other PRs and makes the develop branch fail CI.

**Comment by kain88-de** [comment]
somehow filtering warnings under python2 with the warnings contextmanager doesn't work reliable. I'm now running the testsuite on my machine with python2. I want to see if I can get the same error. Otherwise the PR #1912 has to be reverted and cleaning up the warnings has to wait until we drop support for python 2

**Comment by kain88-de** [comment]
So yeah this passes without any issues on my own machine

**Comment by kain88-de** [comment]
Since this is hard to reproduce and I have no idea where this comes from I would just say we revert #1912 

Anything against that suggestion?

**Comment by orbeckst** [comment]
I kind of hate reverting your thoughtful work in #1912 ; if it were *only* this test, we could just keep ignoring the warning or xfailing it instead of looking for it (with an added comment referring to this issue). If more of these weird failures show up then reverting is a more sane approach.

**Comment by kain88-de** [comment]
this isn't a one off thing. I had other prs trying to get the warnings 
to a reasonable level so we can monitor them also for failures. This was 
good to catch our own deprecations.

But python 2.7 has always been difficult with this and is the reason I 
never managed to turn warnings into errors for our test-suite.

**Comment by orbeckst** [comment]
I don't have much more input – if you think reverting is best then go ahead (and perhaps undelete the branch from which PR #1912 came so that it can be re-used later...).

It is more important to have Python 2.7 tests operational than going through the warnings.

(btw, I am mildly shocked that [your SO question on a Python topic](https://stackoverflow.com/q/46247626) has *zero* answers...)

**Comment by orbeckst** [comment]
PR #1917 should "fix" this (by not testing warnings).
